pkg/proxy/iptables/proxier.go

```diff
@@ -1240,7 +1240,7 @@ func (proxier *Proxier) syncProxyRules() {
 					"-A", string(kubeServicesChain),
 					"-m", "comment", "--comment", fmt.Sprintf(`"%s loadbalancer IP"`, svcNameString),
 					"-m", protocol, "-p", protocol,
-					"-d", utilproxy.ToCIDR(netutils.ParseIPSloppy(ingress)),
+					"-d", ingress,
```

Comment:
hmm , unrelated to the code here, but I just realize that this can be a security hole for the weird IPs parsing thing, the loadbalancer IPs are set by external controllers and the don't do validation, actually we have/jad several bugs because of having trailing spaces
https://github.com/sickcodes/security/blob/master/advisories/SICK-2021-016.md

```
$ iptables -A INPUT -d 10.008.011.012 -m comment --comment test-ip-zeros -j ACCEPT
$ iptables-save | grep test-ip-zero
-A INPUT -d 10.8.9.10/32 -m comment --comment test-ip-zeros -j ACCEPT
``` 

should we parse the IPs and canonicalize them in 
https://github.com/kubernetes/kubernetes/blob/a9ad15c42102ce6a53ece6d3b6f41e85f2d89f60/pkg/proxy/service.go#L113-L120

Reply:
> actually we have/jad several bugs because of having trailing spaces

yeah, there's a test for that in `proxier_test.go`

> should we parse the IPs and canonicalize them in

yes, probably

Reply:
We do validate that `.status.loadBalancer.ingress[].ip` is a IP (albeit sloppy).

I get an error on 008: "host/network `011.012.8.008' not found", while 007 works.

I'm fine to canonicalize them with a big comment saying "this is how iptables parses the octal corner case".   The real problem with this is when different tools interpret "012" differently.  At least iptables and conntrack both treat leading 0 as octal:

```
kubernetes-minion-group-81r5 /home/thockin # conntrack -L -d 10.0.0.1
tcp      6 431993 ESTABLISHED src=10.64.2.3 dst=10.0.0.1 sport=39090 dport=443 src=104.155.184.204 dst=10.128.0.20 sport=443 dport=39090 [ASSURED] mark=0 use=1
conntrack v1.4.4 (conntrack-tools): 1 flow entries have been shown.
kubernetes-minion-group-81r5 /home/thockin # conntrack -L -d 012.0.0.1
tcp      6 431983 ESTABLISHED src=10.64.2.3 dst=10.0.0.1 sport=39090 dport=443 src=104.155.184.204 dst=10.128.0.20 sport=443 dport=39090 [ASSURED] mark=0 use=1
conntrack v1.4.4 (conntrack-tools): 1 flow entries have been shown.
```


BTW, iptables also accepts hex.  LOL

```
kubernetes-minion-group-81r5 /home/thockin # iptables -t nat -A OUTPUT -d 0xff.0xff.0xff.0xff -m comment --comment test-hex -j ACCEPT
kubernetes-minion-group-81r5 /home/thockin # iptables-save -t nat | grep test-hex
-A OUTPUT -d 255.255.255.255/32 -m comment --comment test-hex -j ACCEPT
```

Reply:
ah, we already do sanity-check the IP values, as part of the process of reducing the list down to a single IP family

Reply:
I couldn't check that code, but Is not about sanity check, since we are already using the parser that allows the zeroes, those are allowed.
Is about canonicalize, otherwise people can mean IP1 but iptables understand IP2, if we use the IP stringer that problem is solved.